test/dotnet-msbuild.Tests/GivenDotnetRestoreInvocation.cs

```diff
@@ -18,6 +18,11 @@ public class GivenDotnetRestoreInvocation
         private static readonly string WorkingDirectory = 
             TestPathUtilities.FormatAbsolutePath(nameof(GivenDotnetRestoreInvocation));
 
+        public GivenDotnetRestoreInvocation()
+        {
+            Telemetry.Telemetry.CurrentSessionId = null;
```

Comment:
Hmm, this is sprinkled all over. Can we put it in one place with an explanation?